### PR TITLE
repo-updater: Add debug endpoint to get gitserver repo status

### PR DESCRIPTION
**Pull request by ryanslade**
We expose a debug endpoint that will return, in JSON format, the current status on gitserver
of a repo by name.

For example, locally:

`https://sourcegraph.test:3443/-/debug/proxies/repo-updater-127.0.0.1/gitserver-repo-status?repo=github.com/sourcegraph/sourcegraph`

```json
{
  "RepoID": 16,
  "ShardID": "127.0.0.1:3178",
  "CloneStatus": "cloned",
  "LastError": "",
  "LastFetched": "2022-02-21T16:30:18.785482Z",
  "LastChanged": "2022-02-21T16:30:20.201032Z",
  "UpdatedAt": "2022-02-21T16:30:20.272283Z"
}
```

This should make it easier for us to get this information from customers and customer support
without dropping down to the DB.

# Test plan

Tested manually on my local instance, see output above.

### CODE REVIEW

File: cmd/repo-updater/shared/main.go [hunk 1]
```diff
@@ -330,7 +340,29 @@ func Main(enterpriseInit EnterpriseInit) {
...
+	debugserverEndpoints.gitserverReposStatus = func(w http.ResponseWriter, r *http.Request) {
+		repo := r.FormValue("repo")
+		if repo == "" {
+			http.Error(w, "missing 'repo' param", http.StatusBadRequest)
+			return
+		}
+
+		status, err := db.GitserverRepos().GetByName(r.Context(), api.RepoName(repo))
+		if err != nil {
+			http.Error(w, fmt.Sprintf("fetching status: %q", err), http.StatusInternalServerError)
```
Comment (sashaostrikov) [inline cmd/repo-updater/shared/main.go:354]:
```suggestion
			http.Error(w, fmt.Sprintf("fetching repository status: %q", err), http.StatusInternalServerError)
```

File: internal/database/gitserver_repos.go [hunk 2]
```diff
@@ -285,7 +286,29 @@ FROM gitserver_repos
 WHERE repo_id = %s
 `
 
-	row := s.QueryRow(ctx, sqlf.Sprintf(q, id))
+	return scanSingleGitserverRepo(s.QueryRow(ctx, sqlf.Sprintf(q, id)))
+}
+
+func (s *gitserverRepoStore) GetByName(ctx context.Context, name api.RepoName) (*types.GitserverRepo, error) {
+	q := `
+-- source: internal/database/gitserver_repos.go:gitserverRepoStore.GetByName
```
Comment (sashaostrikov) [inline internal/database/gitserver_repos.go:294]: just a question: why do we add such comments? is this used to trace sources of DB queries?

**Review by sashaostrikov — APPROVED** [review]
LGTM!

File: internal/database/gitserver_repos.go (hunk 2 above)
Comment (ryanslade) [reply internal/database/gitserver_repos.go:294]: Yep, that's exactly it. It's so that when you see SQL in a trace you can easily find it in code.

**Pull request merged**